### DISCUSSION TITLE
wire:loading not working on wire:click element

### DISCUSSION THREAD

**Discussion by gjm** [livewire/livewire#7353] [1. Help]
Hi.

I have a link that dispatches an event and want to toggle a class using `wire:loading.class` but it's not working. When I click the link the class doesn't toggle.

```
<a href="#" wire:click.prevent="$dispatch('event')" wire:loading.class="disabled">click me</a>
```

Tried placing the exact same link next to it, without the `wire:click` attribute and it toggles the class. 

Is there some caveat on using `wire:loading` on the element that triggers the request?

Thanks in advance.

**Comment by whousehomestay** [comment]
Maybe my question is related?
https://github.com/livewire/livewire/discussions/7287
